lib/http/cookie_jar/yaml_saver.rb

```diff
@@ -73,4 +73,14 @@ def load(io, jar)
   def default_options
     {}
   end
+
+  if YAML.name == 'Psych' && Psych::VERSION >= '3.1'
+    def load_yaml(yaml)
+      YAML.safe_load(yaml, :permitted_classes => %w[Time HTTP::Cookie Mechanize::Cookie])
```

Comment:
psych >=3.1 supports keyword arguments.  We still support ruby 1.8.7, hence the rocket syntax.